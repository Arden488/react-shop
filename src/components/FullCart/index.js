import React from 'react';

const FullCart = () => (
  <div>FullCart</div>
);

export default FullCart;
