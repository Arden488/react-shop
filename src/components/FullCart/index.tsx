import * as React from 'react';

const FullCart = () => (
  <div>FullCart</div>
);

export default FullCart;
