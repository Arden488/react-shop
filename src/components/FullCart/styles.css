.full-cart-item {
  display: grid;
  grid-template-columns: 200px auto 300px;
  grid-column-gap: 20px;
  border-bottom: 1px solid #e6e6e6;
  padding-bottom: 20px;
  margin-bottom: 20px;
}

.full-cart-item__image {
  max-width: 100%;
}

.full-cart-item__controls {
  display: grid;
  grid-template-columns: 150px 100px 50px;
}

.full-cart__summ {
  padding: 20px 0;
  max-width: 400px;
  text-align: center;
  margin: auto;
}

.full-cart__summ-value {
  font-weight: bold;
  margin-left: 30px;
  font-size: 24px;
}

.full-cart__proceed {
  display: block;
  padding: 10px;
  max-width: 400px;
  width: 100%;
  margin: auto;
}
