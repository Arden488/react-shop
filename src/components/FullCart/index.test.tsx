/* global it, expect, shallow, beforeEach */

import React from 'react';

import FullCart from './index';

let component = null;

beforeEach(() => {
  component = shallow(<FullCart />);
});

it('renders correctly', () => {
  expect(component).toMatchSnapshot();
});
