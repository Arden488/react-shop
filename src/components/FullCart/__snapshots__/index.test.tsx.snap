// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Cart
  </h1>
  <div
    className="full-cart"
  >
    <div
      className="full-cart-item"
    >
      <img
        alt="Apple iPhone 8 64 ГБ (PRODUCT)RED"
        className="full-cart-item__image"
        src="/img/iphone8_red.webp"
      />
      <div
        className="full-cart-item__title"
      >
        Apple iPhone 8 64 ГБ (PRODUCT)RED
      </div>
      <div
        className="full-cart-item__controls"
      >
        <div
          className="full-cart-item__qnt"
        >
          Quantity
        </div>
        <div
          className="full-cart-item__price"
        >
          39 990 ₽
        </div>
        <div
          className="full-cart-item__remove"
        >
          <button>
            Remove
          </button>
        </div>
      </div>
    </div>
    <div
      className="full-cart-item"
    >
      <img
        alt="Apple iPhone 8 64 ГБ (PRODUCT)RED"
        className="full-cart-item__image"
        src="/img/iphone8_red.webp"
      />
      <div
        className="full-cart-item__title"
      >
        Apple iPhone 8 64 ГБ (PRODUCT)RED
      </div>
      <div
        className="full-cart-item__controls"
      >
        <div
          className="full-cart-item__qnt"
        >
          Quantity
        </div>
        <div
          className="full-cart-item__price"
        >
          39 990 ₽
        </div>
        <div
          className="full-cart-item__remove"
        >
          <button>
            Remove
          </button>
        </div>
      </div>
    </div>
    <div
      className="full-cart-item"
    >
      <img
        alt="Apple iPhone 8 64 ГБ (PRODUCT)RED"
        className="full-cart-item__image"
        src="/img/iphone8_red.webp"
      />
      <div
        className="full-cart-item__title"
      >
        Apple iPhone 8 64 ГБ (PRODUCT)RED
      </div>
      <div
        className="full-cart-item__controls"
      >
        <div
          className="full-cart-item__qnt"
        >
          Quantity
        </div>
        <div
          className="full-cart-item__price"
        >
          39 990 ₽
        </div>
        <div
          className="full-cart-item__remove"
        >
          <button>
            Remove
          </button>
        </div>
      </div>
    </div>
    <div
      className="full-cart__summ"
    >
      Summary:
      <span
        className="full-cart__summ-value"
      >
        39 990 ₽
      </span>
    </div>
    <button>
      <Link
        replace={false}
        to="/checkout"
      >
        Proceed to checkout
      </Link>
    </button>
  </div>
</div>
`;
